### PR TITLE
[Storage] [Typing] [File Share] Decoupled `_file_client.py` and `_file_client_async.py`

### PR DESCRIPTION
**Pull request by weirongw23-msft**
This PR contains these changes:

- Annotated class APIs
- Decoupled `_file_client.py` and `_file_client_async.py`
- Removed `# type: ignore` comments
- Formatted `# pylint: disable=...` to our existing convention
- Refactored URL and snapshot parsing logic
- Refactored `options` code
- Reordered imports

### CODE REVIEW

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/_file_client.py [hunk 1]
```diff
@@ -53,22 +59,23 @@
...
+    max_concurrency,
+    file_settings,
+    file_attributes="none",
+    file_creation_time="now",
+    file_last_write_time="now",
+    file_permission=None,
+    file_permission_key=None,
+    progress_hook=None,
+    **kwargs
+):
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/_file_client.py:78]: Doesn't have to be this PR, but you'll need to annotate these types eventually.

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/_file_client_helpers.py [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+def _parse_snapshot(
+    snapshot: Optional[Union[str, Dict[str, Any]]] = None,
+    path_snapshot: Optional[str] = None
+) -> Optional[str]:
+    if hasattr(snapshot, 'snapshot'):
+        return snapshot.snapshot
+    if isinstance(snapshot, Dict):
+        return snapshot['snapshot']
+    return snapshot or path_snapshot
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/_file_client_helpers.py:44]: Hmm I think we have repeated this in a couple of helpers now. Maybe we can move it to somewhere else and call the same one from all clients? Perhaps the `parsers.py` file and then we can remove it from all the helper files and just use the common one.

File: sdk/storage/azure-storage-file-share/azure/storage/fileshare/_file_client_helpers.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+def _from_file_url(
+    file_url: str,
+    snapshot: Optional[Union[str, Dict[str, Any]]] = None
+) -> Tuple[str, str, str, Optional[Union[str, Dict[str, Any]]]]:
+    try:
+        if not file_url.lower().startswith('http'):
+            file_url = "https://" + file_url
+    except AttributeError as exc:
+        raise ValueError("File URL must be a string.") from exc
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-file-share/azure/storage/fileshare/_file_client_helpers.py:55]: Let's replace this try/catch with an `isinstance(file_url, str)` and throw in the else.

**Review by vincenttran-msft — APPROVED** [review]
LGTM 😄

**Review by jalauzon-msft — APPROVED** [review]

**Pull request merged**